ortools/linear_solver/xpress_interface.cc

```diff
@@ -1148,15 +1148,17 @@ void XpressInterface::ExtractNewVariables() {
     unique_ptr<double[]> lb(new double[newcols]);
     unique_ptr<double[]> ub(new double[newcols]);
     unique_ptr<char[]> ctype(new char[newcols]);
-    unique_ptr<const char*[]> colname(new const char*[newcols]);
+    std::vector<char> colname;
 
     bool have_names = false;
     for (int j = 0, varidx = last_extracted; j < newcols; ++j, ++varidx) {
       MPVariable const* const var = solver_->variables_[varidx];
       lb[j] = var->lb();
       ub[j] = var->ub();
       ctype[j] = var->integer() ? XPRS_INTEGER : XPRS_CONTINUOUS;
-      colname[j] = var->name().empty() ? 0 : var->name().c_str();
+      std::copy(var->name().begin(), var->name().end(),
+                std::back_inserter(colname));
+      colname.push_back('\0');
```

Comment:
XPRESS expects a long string such as this one "name1\0name2\0name3". A bit unusual for a C API where you'd expect a goold old `const char**`